packages/playout-gateway/src/tsrHandler.ts

```diff
@@ -227,24 +234,97 @@ export class TSRHandler {
 				this.logger.error('Error in setTimelineTriggerTime', e)
 			})
 		})
-		this.tsr.on('timelineCallback', (time, objId, callbackName, data) => {
-			// @ts-expect-error Untyped bunch of methods
-			const method = PeripheralDeviceAPIMethods[callbackName]
-			if (method) {
-				this._coreHandler.core
-					.callMethod(method, [
-						Object.assign({}, data, {
-							objId: objId,
-							time: time,
-						}),
-					])
-					.catch((e) => {
-						this.logger.error('Error in timelineCallback', e)
-					})
-			} else {
-				this.logger.error(`Unknown callback method "${callbackName}"`)
+
+		let changedResults: PlayoutChangedResults | undefined = undefined
+		let sendCallbacksTimeout: NodeJS.Timer | undefined = undefined
+
+		this.tsr.on(
+			'timelineCallback',
+			(time, objId, callbackName0, data: PartPlaybackCallbackData | PiecePlaybackCallbackData) => {
```

Comment:
`init` got quite long. Can implementations of these callbacks be extracted out of it?